lib/sqlalchemy/ext/horizontal_shard.py

```diff
@@ -449,7 +452,7 @@ def execute_and_instances(
 
     def iter_for_shard(
         shard_id: ShardIdentifier,
-    ) -> Union[Result[_T], IteratorResult[_TP]]:
```

Comment:
**Michael Bayer** (zzzeek) wrote:

i think this should be just Result[Unpack[_Ts]] now, there's no need to union it with itself.   pylance currently doesnt like the invoke_statement() return with the union.

update: it has to be TupleAny for now, this is fine, this whole fn is an event handler it's not public facing

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/4986

Reply:
Done